cmd/allocator/main.go

```diff
@@ -275,7 +309,9 @@ func getCACertPool(path string) (*x509.CertPool, error) {
 type serviceHandler struct {
 	allocationCallback func(*allocationv1.GameServerAllocation) (k8sruntime.Object, error)
 	certMutex          sync.RWMutex
```

Comment:
It's best to put the mutex next to the thing that it guards, so that it's obvious what is protected by the mutex. So:

type serviceHandler struct {
  allocationCallback ...

  certMutex sync.RWMutex
  caCertPool *x509.CertPool

  tlsMutex sync.RWMutex
  tlsCert *tls.Certificate
}

Reply:
Reorganized the members in (https://github.com/googleforgames/agones/pull/1638).